### PR TITLE
Topic recognition - Khai Nguyen Tri 46977016

### PR DESCRIPTION
**Pull request by buffwarren**
Author: Khai Nguyen Tri

Student Number: s4697701 / 46977016

Description: Spec 1 in assignment documentation:
1. Segment the ISIC 2017/8 data set with the Improved UNet [1] with all labels having a minimum Dice similarity coefficient of 0.8 on the test set.

The model was able to achieve a DCE of >0.8 on the test set with the Improved U-Net model. 30 epochs were used for training.

**File Structure:**

**modules.py**: contains the source code of the components of the Improved U-Net model. Each component is be implemented as a class or a function

**dataset.py**: containing the data loader for loading and preprocessing the 2017 ISIC Dataset

**train.py**: containing the source code for training, validating, testing and saving the Improved U-Net model. Model is imported from “modules.py” and the data loader is imported from “dataset.py”. Losses & metrics plotted and documented throughout. Checkpointing implemented and model saved systematically.

**predict.py**: shows example usage of the Improved U-Net model from a saved model produced by train.py. Results, metrics and visualisation also implemented.

**README.MD**: documents entire model. Dependencies & references here.

The plots folder also shows visualisations of model performance, architecture and results

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files, but wrong readme.md place.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.6579
- Correctly using PyTorch to construct the Improved U-Net models and functions, but didn't find data augmentation.
- Good coding design.
- Sufficient docstring and comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by buffwarren** [comment]
Hi, I was just wondering what the data augmentation was referring to?

Also, I think I made a typo when reporting the average dice coefficient as it was over 0.8 if memory serves me right when I ran it with my epoch 30 model, any chance I could correct it?

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem but could use augmentation to improve performance -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required (update the README location and revert repo README.) -2
Request Description good

**Comment by shakes76** [comment]
Cannot merge because the repo readme has been modified and PR readme is in the wrong place (is the repo readme and should be in your solution folder). Once fixed I can merge and award the feedback marks.

**Comment by buffwarren** [comment]
Hey, I just moved the README location and also restored the deleted ones in the outer directories. Let me know if there are any further issues.

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2
